### ISSUE TITLE
null is not an object (evalulating 'MediaManager.play')

### ISSUE THREAD

**Issue by OwenMelbz** [arielfaur/ionic-audio#60]
I've started a fresh Ionic v1 App, and have effectively lifted your demo examples. I'm currently testing it on an iOS iPhone 6 emulator, and an iPhone 5 with iOS 9

(I've bundled up the whole working directory in a zip as that might be useful https://dl.dropboxusercontent.com/u/46252170/nrf1.zip)

p.s. Is there a way to compile your version 2 written in typescript into a standalone js file which can be included and used?

This is the HTML loop I have

```html
    <div class="list">
      <ion-audio-track ng-repeat="track in tracks" track="track">
        <div class="card">
          <div class="item item-thumbnail-left">
            <img src="{{track.art}}">

            <h2>{{track.title}}</h2>

            <p>{{track.artist}}</p>
            <ion-audio-controls>
              <a class="button button-icon icon" ion-audio-play></a>
              <ion-spinner icon="ios" style="position: relative; top: 8px; left: 4px"></ion-spinner>
            </ion-audio-controls>
          </div>
          <div class="item item-divider">
            <ion-audio-progress-bar display-time></ion-audio-progress-bar>
          </div>
        </div>
      </ion-audio-track>
    </div>
```

this is my controller

```js
.controller('DashCtrl', function($scope) {

    $scope.tracks = [
        {
            url: 'https://ionic-audio.s3.amazonaws.com/Message%20in%20a%20bottle.mp3',
            artist: 'The Police',
            title: 'Message in a bottle',
            art: 'https://ionic-audio.s3.amazonaws.com/The_Police_Greatest_Hits.jpg'
        },
        {
            url: 'https://ionic-audio.s3.amazonaws.com/Roxane.mp3',
            artist: 'The Police',
            title: 'Roxane',
            art: 'https://ionic-audio.s3.amazonaws.com/The_Police_Greatest_Hits.jpg'
        }
    ];


})
```

Here is a list of the cordova plugins i installed using `cordova plugin add cordova-plugin-media`

```sh
Saved api_key, writing to ionic.io.bundle.min.js...
owen@batman:~/nrf1$ cordova plugins list
cordova-plugin-compat 1.0.0 "Compat"
cordova-plugin-console 1.0.3 "Console"
cordova-plugin-device 1.1.2 "Device"
cordova-plugin-file 4.2.0 "File"
cordova-plugin-media 2.2.0 "Media"
cordova-plugin-splashscreen 3.2.2 "Splashscreen"
cordova-plugin-statusbar 2.1.3 "StatusBar"
cordova-plugin-whitelist 1.2.2 "Whitelist"
ionic-plugin-keyboard 2.2.1 "Keyboard"
```

Yet when I click play - I inspect the index.html via Safari and it returns the following error

![web_inspector_ _simulator_ _nrf1_ __null__ _index_html_and_favorites_and_ios_support_ _issue__19_ _arielfaur_ionic-audio](https://cloud.githubusercontent.com/assets/1094740/16619311/84bd7c4a-4385-11e6-9877-80fd48b5d336.png)

How could I go about getting this issue resolved please?
Many thanks

p.s. Not sure what else is needed for some help but here is some other configs

ionic info
```sh
Your system information:

Cordova CLI: 6.2.0
Gulp version:  CLI version 3.9.0
Gulp local:   Local version 3.9.1
Ionic Framework Version: 1.3.1
Ionic CLI Version: 1.7.16
Ionic App Lib Version: 0.7.3
ios-deploy version: 1.8.6 
ios-sim version: 5.0.8 
OS: Mac OS X El Capitan
Node Version: v4.2.4
Xcode version: Xcode 7.3.1 Build version 7D1014 
```

bower.json
```js
{
  "name": "HelloIonic",
  "private": "true",
  "devDependencies": {
    "ionic": "driftyco/ionic-bower#1.3.1",
    "ionic-platform-web-client": "^0.7.1"
  },
  "dependencies": {
    "sweetalert": "^1.1.3",
    "ionic-material-design-lite": "^0.5.5",
    "ionic-audio": "^1.3.0"
  },
  "resolutions": {
    "angular-animate": "1.5.3",
    "angular": "1.5.3",
    "angular-sanitize": "1.5.3",
    "ionic": "1.3.1"
  }
}
```

package.json
```js
{
  "name": "HelloIonic",
  "private": "true",
  "devDependencies": {
    "ionic": "driftyco/ionic-bower#1.3.1",
    "ionic-platform-web-client": "^0.7.1"
  },
  "dependencies": {
    "sweetalert": "^1.1.3",
    "ionic-material-design-lite": "^0.5.5",
    "ionic-audio": "^1.3.0"
  },
  "resolutions": {
    "angular-animate": "1.5.3",
    "angular": "1.5.3",
    "angular-sanitize": "1.5.3",
    "ionic": "1.3.1"
  }
}

```

config.xml
```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<widget id="com.ionicframework.nrf1534241" version="0.0.1" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0">
  <name>nrf1</name>
  <description>
        An Ionic Framework and Cordova project.
    </description>
  <author email="you@example.com" href="http://example.com.com/">
      Your Name Here
    </author>
  <content src="index.html"/>
  <access origin="*"/>
  <preference name="webviewbounce" value="false"/>
  <preference name="UIWebViewBounce" value="false"/>
  <preference name="DisallowOverscroll" value="true"/>
  <preference name="SplashScreenDelay" value="2000"/>
  <preference name="FadeSplashScreenDuration" value="2000"/>
  <preference name="android-minSdkVersion" value="16"/>
  <preference name="BackupWebStorage" value="none"/>
  <feature name="StatusBar">
    <param name="ios-package" value="CDVStatusBar" onload="true"/>
  </feature>
</widget>
```

index.html
```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="initial-scale=1, maximum-scale=1, user-scalable=no, width=device-width">
    <title></title>


    <!-- compiled css output -->
    <link href="css/ionic.app.css" rel="stylesheet">

    <!-- ionic/angularjs js -->
    <script src="lib/ionic/js/ionic.bundle.js"></script>
    <script src="lib/ionic-platform-web-client/dist/ionic.io.bundle.min.js"></script>

    <!-- cordova script (this will be a 404 during development) -->
    <!-- Cordova is bootstrapped by ionic-platform-web-client, uncomment this if you remove ionic-platform-web-client... -->
<!-- <script src="cordova.js"></script> -->

    <script src="lib/sweetalert/dist/sweetalert.min.js"></script>
    <script src="lib/ionic-audio/dist/ion-audio.js"></script>

    
    <!-- your app's js -->
    <script src="js/app.js"></script>
    <script src="js/controllers.js"></script>
    <script src="js/services.js"></script>
  </head>
  <body ng-app="starter">
    <!--
      The nav bar that will be updated as we navigate between views.
    -->
    <ion-nav-bar class="bar-stable">
      <ion-nav-back-button>
      </ion-nav-back-button>
    </ion-nav-bar>
    <!--
      The views will be rendered in the <ion-nav-view> directive below
      Templates are in the /templates folder (but you could also
      have templates inline in this html file if you'd like).
    -->
    <ion-nav-view></ion-nav-view>
  </body>
</html>

```

**Comment by OwenMelbz** [comment]
Update:

I've been having a poke around, and it seems there might be a race condition issue - Effectively I ran the app several times, 80% of the time it will fail with the error, 20% of the time it works, all I'm doing is running `ionic emulate ios` - so considering that no code is changing - either ionic is doing something weird at the compilation time that is sometimes failing - or there is a race condition issue that is causing it.

Not sure however how to track the issue down though, but ideas would be just lovely!

**Comment by arielfaur** [comment]
Sorry for the delay, I was on vacation :smile: 

Have you tried updating Cordova Media plugin to the latest version (2.3.0) ? In many cases this solves some of the issues. Also, does it work on your iOS device? I don't trust the emulator sometimes, especially when working with Cordova plugins.

**Comment by arielfaur** [comment]
@OwenMelbz  any updates on this? I recently updated the plugin and fixed a minor issue. Also, I updated my cordova version to latest (including the media plugin) and it is supposed to solve some issues with playback. Let me know.

**Comment by OwenMelbz** [comment]
@arielfaur Hi sorry,

I do not know exactly - however I've resolved it.

I believe the issue was me using the latest versions of AngularJS and its dependencies - When a conflict came up during `bower install` i picked the latest versions of everything.

When i tried again on another install, I picked the version that `ionic-audio` needed - and this worked.

So I believe the issue was todo with the later versions of AngularJS

The resolutions i picked were

```json
"resolutions": {
    "angular-animate": "1.5.3",
    "angular": "1.5.3",
    "angular-sanitize": "1.5.3",
    "ionic": "1.3.1"
  }
```
